src/xml_threat_model_reviewer2.py

```diff
@@ -0,0 +1,187 @@
+    ImmediateExecutorCapability().add_to_agent(answerer_agent, add_answer,
+                                               description="Add an answer to a spec question")
+
+    assistant = AssistantAgent(
+        name="Threat_Model_Evaluator",
+        description="An agent that evaluates the quality of a threat model.",
+    )
+
+    def summarize(self, recipient, summary_args):
+        spec_answers: List[Tuple[Spec, SpecAnswer]] = []
+        for spec_id, spec_answer in cap.spec_index_to_answer.items():
+            spec_question = next(filter(lambda x: x.id == spec_id, specs))
+            spec_answers.append((spec_question, spec_answer))
+        spec_answers = sorted(spec_answers, key=lambda x: x[1].spec_id)
+        
+        containers = [build_container_for_answer(spec, spec_answer) for spec, spec_answer in spec_answers]
+        card = build_adative_card(containers)
+        def set_default(obj):
+            if isinstance(obj, set):
+                return list(obj)
+            raise TypeError
+        return f"adaptive_card:{json.dumps(card, ensure_ascii=False, default=set_default)}"
+
+    assistant.register_nested_chats([
+        {
+            "recipient": questioner_agent,
+            "sender": answerer_agent,
+            "summary_method": summarize,
+            "chat_id": 1,
+        },
+    ], trigger=lambda sender: sender not in [assistant], use_async=True)
+
+    return assistant
+
+def build_container_for_answer(spec: Spec, spec_answer: SpecAnswer):
+    return {
+            "type": "Container",
+            "items": [
+                {
+                    "type": "TextBlock",
+                    "text": spec.spec,
+                    "wrap": True,
+                    "weight": "Bolder",
+                },
+                {
+                    "type": "ColumnSet",
+                    "columns": [
+                        {
+                            "type": "Column",
+                            "width": "auto",
+                            "items": [
+                                {
+                                    "type": "TextBlock",
+                                    "text": tag_with_headers[spec_answer.tag],
+                                    "wrap": True
+                                }
+                            ]
+                        },
+                        {
+                            "type": "Column",
+                            "width": "stretch",
+                            "items": [
+                                {
+                                    "type": "TextBlock",
+                                    "text": spec_answer.detailed_answer,
+                                    "wrap": True
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ],
+            "separator": True
+        }
+    
+def build_adative_card(spec_answer_containers: List[Dict]):
```

Comment:
nit: adaptive (spelling)

The general downside of using adaptive cards directly within the bot response is that it disallows multiple views for the output. If trust copilot, for example, wants to have an HTML view, then it does not have the ability to render an adaptive card necessarily.

Reply:
we can send back json, and have the channel translate it to when sending it back too

Reply:
can be iterative

Reply:
I'm generally not a huge fan of this approach, but it's something quick n dirty and we can iterate to build on top